src/lancode.c

```diff
@@ -284,8 +284,7 @@ int send_lan_message(int opcode, char *message) {
     }
 
     if (opcode == LOGENTRY) {
-	sendbuffer[82] = '\0';
-
+	strcat(sendbuffer, "\n");
```

Comment:
sendbuffer seems to contain a correctly formatted log line. I tested it OK without adding a new line.

Reply:
The reason for adding the newline is on the receiving end. Lines 137-139 in background_process.c always drop the last character of the received string. That is the normal code used to drop appended newlines from strings. 

As all LAN-messages are sent as normal text strings it seems appropriate to terminate any of them with a newline. That also makes the debuglog more readable.

I think we should harmonize all LAN messages to that behavior. That would also allow us to simplify the send_lan_message() code.

At the moment (with the suggested changes) we add a newline to almost all messages. Only exception is send_time() where a space is added to the time string to avoid truncation. That should be change too.

Reply:
We are processing our own messages, so any preprocessing is at least suspicious. Blindly dropping the last char is even worse. This is a legacy code originating from af43a4d. It must have been a quick, but incorrect fix for some unknown problem.

Logging shall cope with any message and in fact it does this OK in background_process.c:
```
    148                     fputs(lan_message, fp);
    149                     fputs("\n", fp);
```
So I see no reason adding NL on the sending side.

Reply:
Yes, the whole code (besides the QTCxxx messages) are ancient code (It is also in the very early 0.9.10 version from 2003). There was neither time nor need to sort it out. But let us do it now.

I do not think that it was a fix - I assume Rein had NL terminated strings in mind when writing these lines. But the whole artificial fixes of line length (lan_loglin[xx] = '\0') looks really like hot fixes, especially as all these xx values are mostly wrong.

In meantime I did some work to harmonize the LAN message format to EVERY time use of at least ONE NL (which would be dropped by the two lines in question) - see last commit. The generated messages sometimes contains two NL (even without my change). So that mess should really be sorted out.

See my second message as comment to your plan below.

Reply:
Could the removal of last char be made more relaxed using `g_strchomp()` instead?